// db/index.js
module.exports = {
    conns: {
    },
    getLink: function(alias) {
        var me = this;

        alias = alias || 'default';
        var connData = config.databases[alias];

        if (connData) {
            switch (connData['type'] || 'mysql') {
                case 'mysql':
                    if (!me.conns[alias]) {
                        me.conns[alias] = require('mysql').createPool({
                            host: connData.host,
                            user: connData.user,
                            password: connData.pass,
                            database: connData.name
                        });
                    }

                    me.conns[alias].query = function() {
                        var args = arguments;
                        return me.conns[alias].getConnection(function(err, conn) {
//                            conn.config.queryFormat = function(query, values) {
//                                if (!values)
//                                    return query;
//                                return query.replace(/\:(\w+)/g, function(txt, key) {
//                                    if (values.hasOwnProperty(key)) {
//                                        return this.escape(values[key]);
//                                    }
//                                    return txt;
//                                }.bind(this));
//                            };

                            conn.query.apply(conn, args);
                            conn.end();
                        });
                    };

                    return me.conns[alias];
                    break;
            }
        }

        return null;
    }
};
